pgjdbc/src/main/java/org/postgresql/core/PGStream.java

```diff
@@ -372,9 +362,10 @@ public void sendInteger2(int val) throws IOException {
     if (val < 0 || val > 65535) {
       throw new IllegalArgumentException("Tried to send an out-of-range integer as a 2-byte unsigned int value: " + val);
     }
-    int2Buf[0] = (byte) (val >>> 8);
-    int2Buf[1] = (byte) val;
-    pgOutput.write(int2Buf);
+    byte[] buffer = new byte[2];
+    buffer[0] = (byte) (val >> 8);
+    buffer[1] = (byte) val;
+    pgOutput.write(buffer);
```

Comment:
Why making this change?

Reply:
to buffer bytes before writing as pgOutput is always buffered inside PGStream unless its a character. (that's how it is written everytime within PGStream

Effectively reduces the number of IO operations. Not sure if it helps, because bufferSize of pgOutput is always big.